test/Microsoft.AspNetCore.Routing.Tests/DefaultLinkGeneratorTest.cs

```diff
@@ -98,6 +98,40 @@ public void GetLink_MultipleEndpoints_Success2()
             Assert.Equal("/Home/Index", link);
         }
 
+        [Theory]
+        [InlineData("a/b/c", "/Home/Index/a%2Fb%2Fc")]
+        [InlineData("a/b b1/c c1", "/Home/Index/a%2Fb%20b1%2Fc%20c1")]
+        public void GetLink_EncodesValue_OfSingleAsteriskCatchAllParameter(string routeValue, string expected)
+        {
+            // Arrange
+            var endpoint = EndpointFactory.CreateMatcherEndpoint("{controller}/{action}/{*path}");
+            var linkGenerator = CreateLinkGenerator(endpoint);
+            var httpContext = CreateHttpContext(ambientValues: new { controller = "Home", action = "Index" });
+
+            // Act
+            var link = linkGenerator.GetLink(httpContext, new { path = routeValue });
+
+            // Assert
+            Assert.Equal(expected, link);
+        }
+
+        [Theory]
+        [InlineData("a/b/c", "/Home/Index/a/b/c")]
+        [InlineData("a/b b1/c c1", "/Home/Index/a/b b1/c c1")]
+        public void GetLink_DoesNotEncodesValue_OfDoubleAsteriskCatchAllParameter(string routeValue, string expected)
+        {
+            // Arrange
+            var endpoint = EndpointFactory.CreateMatcherEndpoint("{controller}/{action}/{**path}");
+            var linkGenerator = CreateLinkGenerator(endpoint);
+            var httpContext = CreateHttpContext(ambientValues: new { controller = "Home", action = "Index" });
+
+            // Act
+            var link = linkGenerator.GetLink(httpContext, new { path = routeValue });
+
+            // Assert
+            Assert.Equal(expected, link);
+        }
+
```

Comment:
Can you add some cases where there are characters other than `/` that need to be url-encoded?

Reply:
Some extra tests:
* multiple slashes in a row, e.g. localhost/test//blah
* catchall ends with a trailing slash, e.g. localhost/text/thisiscatchallcontent/

Reply:
yeah, i have those scenarios covered here actually